js_modules/dagit/packages/core/src/assets/usePartitionHealthData.tsx

```diff
@@ -469,14 +467,47 @@ export function usePartitionHealthData(assetKeys: AssetKey[], assetLastMateriali
       ]);
     };
     run();
-  }, [client, missingKeyJSON, assetLastMaterializedAt]);
+  }, [client, missingKey, assetLastMaterializedAt]);
 
   return React.useMemo(() => {
     const assetKeyJSONs = JSON.parse(assetKeyJSON);
     return result.filter((r) => assetKeyJSONs.includes(JSON.stringify(r.assetKey)));
   }, [assetKeyJSON, result]);
 }
 
+/**
+ *  Like usePartitionHealthData, but fetches queries in parallel and allows refetching them all in parallel
+ * @returns
+ */
+export function usePartitionHealthData2(assetKeys: AssetKey[], cacheKey: string) {
```

Comment:
I think we could make the main `usePartitionHealthData` run more queries in parallel but I don't think we want to run them ALL in parallel because the user can actually select hundreds of assets when opening the launch backfill modal. (If you don't make any selection at all, the modal is opened for all your assets), and I think this will flood the backend with pretty expensive requests for asset health data. 

It is a bummer that the serial loading is slow, but we might have to fix that separately.

I think if you want, you can pass your `cacheKey` to the original implementation as a fake `assetLastMaterializedAt` and that will bust the cache the same way. (Could also rename that param to cacheKey because I think that explains what it does nicely!)